src/Sulu/Component/Content/Repository/ContentRepository.php

```diff
@@ -0,0 +1,464 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content\Repository;
+
+use Jackalope\Query\QOM\PropertyValue;
+use Jackalope\Query\Row;
+use PHPCR\Query\QOM\QueryObjectModelFactoryInterface;
+use PHPCR\SessionInterface;
+use PHPCR\Util\QOM\QueryBuilder;
+use Sulu\Component\Content\Compat\LocalizationFinder;
+use Sulu\Component\Content\Compat\StructureType;
+use Sulu\Component\Content\Document\RedirectType;
+use Sulu\Component\DocumentManager\PropertyEncoder;
+use Sulu\Component\Localization\Localization;
+use Sulu\Component\PHPCR\SessionManager\SessionManagerInterface;
+use Sulu\Component\Security\Authentication\UserInterface;
+use Sulu\Component\Webspace\Manager\WebspaceManagerInterface;
+use Symfony\Component\Security\Acl\Exception\Exception;
+
+/**
+ * Content repository which query content with sql2 statements.
+ */
+class ContentRepository implements ContentRepositoryInterface
```

Comment:
currently not fully tested - missing tests:

* permissions
* hasChildren
* linked
* ghost (a bit more tests)